compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2536,6 +2547,8 @@ impl<'a> Parser<'a> {
                         }
                     }
 
+                    // FIXME(eholk): add keyword recovery logic for genness
```

Comment:
I'm considering leaving this for a later PR that's focused on diagnostic improvements around `gen fn`, so that this one can focus on the baseline functionality.

Reply:
Yeah that's fine I guess

Please change the FIXME though to not reference a single person, since this is associated with feature work